src/einsteinpy/symbolic/tensor.py

```diff
@@ -501,7 +496,13 @@ def derivative(self, x):
         TypeError
             If argument ``x`` is of type other than listed above
         """
-        x = _parse_vector(x, self.syms)
+        if isinstance(x, BaseRelativityTensor):
+            if x.config != "u" and x.syms != self.syms:
+                raise ValueError(
```

Comment:
Try to do the conditional block like this:
```
        if isinstance(x, BaseRelativityTensor) and x.config != "u" and x.syms != self.syms:
            raise ValueError( ...
```

Reply:
Looks like it's adding +1 to `and`/`or` operators too. If this is really important, I suggest to drop support for list type argument. It's one line of code on users part to convert it to `BaseRelativityTensor` anyway. What else could we do besides split it up into more functions?